MediaBrowser.Controller/Persistence/IItemRepository.cs

```diff
@@ -7,157 +7,82 @@
-        /// Gets the item list.
-        /// </summary>
-        /// <param name="query">The query.</param>
-        /// <returns>List&lt;BaseItem&gt;.</returns>
-        List<BaseItem> GetItemList(InternalItemsQuery query);
-
-        /// <summary>
-        /// Updates the inherited values.
-        /// </summary>
-        void UpdateInheritedValues();
-
-        int GetCount(InternalItemsQuery query);
-
-        QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetGenres(InternalItemsQuery query);
-
-        QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetMusicGenres(InternalItemsQuery query);
-
-        QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetStudios(InternalItemsQuery query);
-
-        QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetArtists(InternalItemsQuery query);
-
-        QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetAlbumArtists(InternalItemsQuery query);
-
-        QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetAllArtists(InternalItemsQuery query);
-
-        List<string> GetMusicGenreNames();
-
-        List<string> GetStudioNames();
-
-        List<string> GetGenreNames();
-
-        List<string> GetAllArtistNames();
-    }
+    /// <param name="id">The id.</param>
+    /// <returns>BaseItem.</returns>
+    BaseItem RetrieveItem(Guid id);
+
+    /// <summary>
+    /// Gets the items.
+    /// </summary>
+    /// <param name="filter">The query.</param>
+    /// <returns>QueryResult&lt;BaseItem&gt;.</returns>
+    QueryResult<BaseItem> GetItems(InternalItemsQuery filter);
+
+    /// <summary>
+    /// Gets the item ids list.
+    /// </summary>
+    /// <param name="filter">The query.</param>
+    /// <returns>List&lt;Guid&gt;.</returns>
+    IReadOnlyList<Guid> GetItemIdsList(InternalItemsQuery filter);
+
+    /// <summary>
+    /// Gets the item list.
+    /// </summary>
+    /// <param name="filter">The query.</param>
+    /// <returns>List&lt;BaseItem&gt;.</returns>
+    IReadOnlyList<BaseItem> GetItemList(InternalItemsQuery filter);
+
+    /// <summary>
+    /// Updates the inherited values.
+    /// </summary>
+    void UpdateInheritedValues();
+
+    int GetCount(InternalItemsQuery filter);
+
+    QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetGenres(InternalItemsQuery filter);
+
+    QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetMusicGenres(InternalItemsQuery filter);
+
+    QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetStudios(InternalItemsQuery filter);
+
+    QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetArtists(InternalItemsQuery filter);
+
+    QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetAlbumArtists(InternalItemsQuery filter);
+
+    QueryResult<(BaseItem Item, ItemCounts ItemCounts)> GetAllArtists(InternalItemsQuery filter);
+
+    IReadOnlyList<string> GetMusicGenreNames();
+
+    IReadOnlyList<string> GetStudioNames();
+
+    IReadOnlyList<string> GetGenreNames();
+
+    IReadOnlyList<string> GetAllArtistNames();
```

Comment:
Any reasons why the docs got removed?

Reply:
They never had any, you need to scroll down
https://github.com/jellyfin/jellyfin/pull/12798/files/96e4d8ca78f839def6f55e8b71d97a3dc22617b8#diff-aa0d5ef8793a2e1913f0cb16716350633a570a442cb453150c906ad5e2ec5680L143